### DISCUSSION TITLE
How to build a actual grain morphology?

### DISCUSSION THREAD

**Discussion by Yuanguo1994** [neperfepx/neper#186] [Q&A] [answered]
Hi, @rquey 

In the neper mannual, it says that the actual shape of grain could be built via raster tessellation. However, I am still confused how to  build the real actual grain morphology after reading through the mannual reference. For example, there is a 2d picture of microstructure of an alloy below. Grain boundaries are marked by white lines for easy distinguish. Is it possible to build this kind of grain shape or analogous shape in 2D via neper? 
If it is possible, what else I need to do in addition to the image (like digitalised the image? or anything else?) and what kind of command should I input? Could you give me some guidance on that?
![image](https://user-images.githubusercontent.com/79046436/114774303-f571cb80-9d67-11eb-9368-7476a04f99ec.png)

Kind regards

Yuanguo

**Comment by rquey** [comment]
> Is it possible to build this kind of grain shape or analogous shape in 2D via neper?

No, this is way to complex. At best, you could write your image as a tesr, as an orientation map and/or a grain map...

**Comment by rquey** [comment, marked as answer]
What I would do to model this one is to generate a tessellation and merge cells to get the same type of morphologies as you have on your image. It does not seem to want to work in 3D but you could use a thin 3D tess to begin with.

```
$ neper -T -n 30 -transform "mergecell(y<0.5)" -morpho "columnar(z)" -domain "cube(1,1,0.01)"
```
![image](https://user-images.githubusercontent.com/2659376/115010021-332c3c80-9ead-11eb-8d37-fc45e71c65a2.png)

_You_ would have to figure out wich cells to merge...

**Reply by Yuanguo1994** [reply]
Hi, Rquey. Thanks for your reply. It is a good idea to model this by mercells.  According to the neper manul of mergecell, the merge should be match the expression. And the expression is based on the variables shown in Section A 2. I have looked through the variables, there are many variables like x, y, z, coordinates, area and so on. I am confused that whether I could merge cell using the cell id? If it is possible, how could it be achieved?

**Reply by Yuanguo1994** [reply]
![image21](https://user-images.githubusercontent.com/79046436/115066422-668bbd00-9ee7-11eb-939b-b850a86a966b.png)
I could merge the cells at the edge of the domain like image show, but I dont know how to merge the cells at the middle of the domain. If I use expression like '0.5<x<1', it doesnt work.

**Reply by rquey** [reply]
For ids, use `id`. You can use logical operators, as in `"(x>0.5)&&(x<1)"`. Be  aware that, if the cells to merge are not adjacent, you might get in trouble...

**Reply by Yuanguo1994** [reply]
Thanks for your reply, I will keep on doing this.

**Reply by Yuanguo1994** [reply]
> For ids, use `id`. You can use logical operators, as in `"(x>0.5)&&(x<1)"`. Be aware that, if the cells to merge are not adjacent, you might get in trouble...

Hi, Rquey 
I have partly completed the work, but now I need to merge the cells via ID. I am still confused how to write the command?
If I want to merge the adjacent cells 2 and 7 shown in the image. Should I write it in the way like 'transform 'mergecell(id=2,7)''?

Kind regards

Yuanguo
 
![image](https://user-images.githubusercontent.com/79046436/115125322-47f8f500-9fbf-11eb-9a2f-1fceae66706e.png)
